Add timestamps columns to competitions.

AlbertoPdRF opened the pull request:
This is necessary for #2999. Another related PR will come soon.

viroulep commented:
If we're gonna do this, why not add both timestamps used by Rails (`created_at`, `updated_at`)?

viroulep commented on WcaOnRails/db/migrate/20180709220826_add_created_at_column_to_competitions.rb:
```diff
@@ -0,0 +1,5 @@
+class AddCreatedAtColumnToCompetitions < ActiveRecord::Migration[5.2]
+  def change
+    add_column :Competitions, :created_at, :datetime
```
If at some point you need to add both timestamps instead of just `created_at` you can use `add_timestamps :Competitions`.

Also do we want to allow for null values here?

jfly replied:
I think we could use `add_timestamps` here instead: https://apidock.com/rails/ActiveRecord/ConnectionAdapters/SchemaStatements/add_timestamps. This would create both a `created_at`, and an `updated_at`. Currently we don't have any plans for the updated_at, but it's kind of a Rails-y thing do have both of them. The only reason the Competitions table doesn't have both of them is because it existed before we switched to Ruby on Rails.

jfly commented on WcaOnRails/app/controllers/competitions_controller.rb:
```diff
@@ -151,6 +151,8 @@ def create
       flash[:success] = t('competitions.messages.create_success')
       redirect_to edit_competition_path(@competition)
+
+      @competition.update!(created_at: Time.now)
```
I don't think you actually don't need this, Rails should manage the `updated_at` column for you. See http://api.rubyonrails.org/classes/ActiveRecord/Persistence.html#method-i-save.

viroulep replied:
Yep, and it touches `created_at` on creation (http://guides.rubyonrails.org/active_record_basics.html#schema-conventions)

AlbertoPdRF commented:
I think I've addressed the comments. The only thing yet to decide is if we want to keep NULL values for past competitions or if I should populate them with something (`announced_at`?) and don't allow NULL values for the new fields.

jfly commented:
@viroulep, @AlbertoPdRF, and I discussed this offline, and decided that it would be most accurate to leave these new timestamps NULL-able. We can always change this later if it proves to be a pain.

viroulep commented:
@AlbertoPdRF, @jfly: turns out it's a pain for #4102, as I use `updated_at` as a key for caching the file :(
Do we have any incentive to actually keep them null?

If we don't, I'd love to create a migration that sets these null fields to `announced_at`, for instance.

AlbertoPdRF commented:
If it's a pain and provided that we don't break anything, I'm fine with your proposal!

jfly commented:
That seems ok to me! I can imagine people asking questions like "how long after confirming does it take for a competition to get announced?", but that's not applicable here.
